lib/anyway/loaders/yaml.rb

```diff
@@ -10,17 +10,37 @@ module Anyway
   module Loaders
     class YAML < Base
       def call(config_path:, **_options)
-        base_config = trace!(:yml, path: relative_config_path(config_path).to_s) { load_base_yml(config_path) }
+        config = trace!(:yml, path: relative_config_path(config_path).to_s) { load_base_yml(config_path) }
+        return config_with_env(config, config_path) if environmental?(config)
 
-        return base_config unless use_local?
+        mix_local!(config, config_path) if use_local?
+        config
+      end
 
-        local_path = local_config_path(config_path)
-        local_config = trace!(:yml, path: relative_config_path(local_path).to_s) { load_local_yml(local_path) }
+      private
 
-        Utils.deep_merge!(base_config, local_config)
+      def config_with_env(config, config_path)
+        env_config = config[Anyway::Settings.current_environment] || {}
+        mix_local!(env_config, config_path) if use_local?
+        return env_config if Anyway::Settings.default_environmental_key.blank?
+
+        default_config = config[Anyway::Settings.default_environmental_key] || {}
```

Comment:
What purpose of `Anyway::Settings.default_environmental_key` is meant?  
If we specify `<< default` in yaml file, YAML parser loads all keys with this scope to our config anyway.
This variable is helpful only when we dont' use inheritance explicitly and envs is too many.
Do I understand it correct?

Reply:
@skryukov Could you please take a look?

Reply:
It was added in https://github.com/palkan/anyway_config/pull/93

Long story short: YAML anchors don't use deep merge, so we introduced this option to enable deep nested defaults.